cs/datatypes/ref.adoc

```diff
@@ -0,0 +1,222 @@
+== @value
+
+make ref! none
+** cannot make error
+
+make ref! #datatype
+== @datatype
+----
+
+=== Konverze `to`
+
+Datový typ `ref!` podporuje řadu typových konverzí, platných pro typeset `any-string!`. Při konverzi dané hodnoty na `ref!` jsou použity níže popsané transformace - s předsazením znaku `@`.
+
+.Datové typy, konvertovatelné na `ref!`
+[options="header" cols="1,5"]
+|===
+| Datatype | Description
+
+| `any-string!`
+| Obsah je kopírován doslovně.
+
+| `any-list!`
+| Prezentace obsažených hodnot, evokované funkcí `form`, jsou zřetězeny (concatenated) a vnější vymezení je odvrženo.
+
+| `binary!`
+| Je konvertováno na řetězec, obsahuje-li platné kódování v UTF-8.
+
+| `none!`
+| Zakázáno.
+
+| Other
+| Stejně jako u `any-list`.
+
+|===
+
+*Příklady*
+
+----
+to-ref [f /o #o]
+== @foo
+
+to-ref #{626172}
+== @bar
+
+to-ref <qux>
+== @qux
+
+to-ref none
+** cannot MAKE/TO string from: none
+----
+Jsouce částí typesetu `any-string!`, podporují hodnoty `ref!` konverzi na mnoho typů:
+
+.Datové typy, konvertovatelné z `ref!`
+[options="header" cols="3,9"]
+|===
+| Datatype | Description
+
+| `any-list!`
+| Ekvivalentní k uzavření hodnoty `ref!` do kontejneru daného typu.
+
+| `any-string!`
+| Obsah je kopírován doslovně.
+
+| `number!`, `money!`, `time!`, `tuple!`
+| Obecně ekvivalentní ke konverzi na `load`, pokud `ref!` reprezentuje dobře formovaný literál daného typu..
+
+Více podrobností o podporovaných literálových formách lze nalézt v příslušné literatuře.
+
+| `none!`, `unset!`
+| Vrací se hodnota daného datového typu (singleton).
+
+| `logic!`
+| Vrací se logická hodnota `true`, odpovídající sémantice jazyka.
+
+| `binary!`
+| Jako výsledek se vrací binární prezentace v kódování UTF-8.
+
+| `char!`
+| První znak, pokud přítomen, je extrahován.
+
+| Other
+| Zakázáno.
+
+|===
+
+*Příklady*
+
+----
+to-paren @ABC
+== (@ABC)
+
+to-tag @tag
+== <tag>
+
+to-float @+88.7
+== 88.7
+
+to-tuple @1.3.3.7
+== 1.3.3.7
+
+to-none @whatever
+== none
+
+to-logic @this-ref-is-false
+== true
+
+to-binary @matrix
+== #{6D6174726978}
+
+to-char @
+** cannot MAKE/TO char from:
+
+to-char @.:
+== #"."
+----
+
+== Operace
+
+Hodnota typu `ref!` podporuje všechny akce pro řady (kromě `put`, `sort`, `trim`) a všechny komparace spou s `min` a `max`.
+
+*Příklady*
+
+----
+reverse @alucard
+== @dracula
+
+find @haystack "needle"
+== none
+
+find @haystack 'stack
+== @stack
+
+skip @stackoverflow 5
+== @overflow
+
+@this = "this"
+== true
+
+@this = @that
+== false
+
+@that == @that
+== true
+
+min @A @Z
+== @A
+
+form @content
+== "content"
+
+mold/part @catastrophy 4
+== "@cat"
+----
+NOTE: Příkaz `form`, aplikovaný na hodnotu `ref!`, zavrhne  znak `@`.
+
+== Jiné funkce
+
+Funkce, souvisící s datovým typem `ref!` avšak v předchozím textu neuvedené, jsou vypsány níže:
+
+* predikát `ref?` vrací `true`, je-li daná hodnota typu `ref!`;
+* velikost písma: `uppercase` and `lowercase`;
+* conversion of URL-style percent-encoded content: `dehex`.
```

Comment:
That line misses translation; also "and" between "`uppercase`" and "`lowercase`" on the line above.

Reply:
line 220 - opravil, dík
line 221 -  PROSBA:
I am not sure what should this sentence  mean. Could you reformulate it for me? 
In Red help (? dehex) I can read: "Converts URL-style hex encoded (%xx) strings.

Reply:
@Tovim will [this](https://cs.wikipedia.org/wiki/URL_k%C3%B3dov%C3%A1n%C3%AD) help? It's a special type of encoding used for URLs. You can even see it in a link to that wiki page:
```
https://cs.wikipedia.org/wiki/URL_k%C3%B3dov%C3%A1n%C3%AD
```

Reply:
Very nice the "this" help, thank you. I am going to translate the line 221 this way:
* konverze URL v hexadecimálním vyjádření (%xx) na textový řetězec: `dehex`
----
>> dehex {URL_k%C3%B3dov%C3%A1n%C3%AD}
== "URL_kódování"
----